src/mount/acquired_files_last_time_used.h

```diff
@@ -0,0 +1,96 @@
+/*
+   Copyright 2013-2017 Skytechnology sp. z o.o.
+   Copyright 2023      Leil Storage OÜ
+
+   This file is part of SaunaFS.
+
+   SaunaFS is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, version 3.
+
+   SaunaFS is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with SaunaFS. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "common/platform.h"
+
+#include <ctime>
+#include <map>
+#include <mutex>
+#include <set>
+#include <stdexcept>
+
+inline uint32_t gCleanAcquiredFilesPeriod;
+inline uint32_t gCleanAcquiredFilesTimeout;
+
+// Define the AcquiredFileLastTimeUsed struct
+struct AcquiredFileLastTimeUsed {
+	uint32_t inode;
+	uint32_t lastTimeUsed;
+
+	// Constructor
+	AcquiredFileLastTimeUsed(uint32_t inode, uint32_t lastTimeUsed)
+	    : inode(inode), lastTimeUsed(lastTimeUsed) {}
+};
+
+struct CompareByLastTimeUsed {
+	bool operator()(const AcquiredFileLastTimeUsed &acquiredFile1,
+	                const AcquiredFileLastTimeUsed &acquiredFile2) const {
+		return acquiredFile1.lastTimeUsed < acquiredFile2.lastTimeUsed ||
+		       (acquiredFile1.lastTimeUsed == acquiredFile2.lastTimeUsed &&
+		        acquiredFile1.inode < acquiredFile2.inode);
+	}
+};
+
+using AcquiredFilesLastTimeUsed =
+    std::set<AcquiredFileLastTimeUsed, CompareByLastTimeUsed>;
+using InodeToFileMap = std::map<uint32_t, AcquiredFilesLastTimeUsed::iterator>;
+
+inline AcquiredFilesLastTimeUsed acquiredFilesLastTimeUsed;
```

Comment:
`acquiredFilesSet` can be the name for the container.

Reply:
Commented [here](https://github.com/leil-io/saunafs/pull/230#discussion_r1872379910)